### PR TITLE
Restricted queues

### PR DESCRIPTION
**Pull request by rcousens**
Rebase of #96 onto latest master to add restricted queues to run command.

I'm not particularly happy with how I butchered the DQL for cleanUpStaleJobs. Should I convert to a query builder?

### CODE REVIEW

**Comment by rcousens** [comment]
Sorry, didn't mean to push the composer.lock through to the pull request as well. Reverted.

**Comment by rcousens** [comment]
Just fixing this up so the tests pass. The query isn't working at the moment so I'll do it via query builder.

**Comment by rcousens** [comment]
Ok I got the tests passing in my own branch by limiting Travis to PHP 5.5. The composer.lock in your current master is pulling in dependencies that require 5.5+ and the travis.yml is set to 5.3 and 5.4.

I've added 5.5 to travis.

**Comment by rcousens** [comment]
Just 1 failing test:

https://travis-ci.org/schmittjoh/JMSJobQueueBundle/jobs/69302928

Will fix it up tomorrow and then I think this is good to go.

**Comment by rcousens** [comment]
@schmittjoh 

Summary:

I've removed PHP 5.3 and 5.4 from travis.yml and added PHP 5.5 and 5.6.

I've updated composer.lock which was way out of date. Symfony 2.8 is PHP 5.5+ and the composer.json pulls in dependencies that won't install on 5.3 and 5.4. The composer.json was already restricting the PHP version to 5.5 anyway.

I've fixed all the failing travis tests due to a previous commit related to worker name and max length. Remaining failing test is for high concurrency. It passes locally but fails on Travis.

I'm out of ideas as to why it fails on Travis, the file is writable but returns empty when read after the logging command jobs finish.

**Comment by rcousens** [comment]
@schmittjoh I know you're busy, would really appreciate this feature making it in or feedback on what I need to do for it to pass :) It's an open feature request by other users too and if it languishes it'll fall behind again.

I spent 4 hours on cleaning things up and restored all your automated tests!

File: Command/RunCommand.php [hunk 1]
```diff
@@ -375,13 +381,24 @@ private function startJob(Job $job)
      *
      * In such an error condition, these jobs are cleaned-up on restart of this command.
      */
-    private function cleanUpStaleJobs($workerName)
+    private function cleanUpStaleJobs($workerName, $restrictedQueues)
```
Comment (schmittjoh) [inline Command/RunCommand.php]: I think we do not need this here. Each worker name must be unique, so the previous query should work fine, no?

Comment (rcousens) [inline Command/RunCommand.php]: Yes good catch, I'll update that now.

I didn't truly realise the intent of worker name until now.

I am confused as to how cleanUpStaleJobs can work on the workerName = NULL case. How can a job be running without a workerName being set (except for backwards compatibility)?

Also, in the case that worker name is not passed as an argument and instead generated from hostname + PID, how do stale jobs get cleaned up if for example an instance of the run command crashes (assume bizarre random segfault etc) and restarted instance has new PID?

Comment (schmittjoh) [inline Command/RunCommand.php]: Yes, ``workerName`` can only be null when running an older version of this bundle, and then migrating to the newer version.

If you use a pre-determined worker-name, stale jobs will be detected faster, i.e. as soon as the Run command is re-started. If you use the default, the CleanUp command will eventually collect stale jobs, but that will be a bit slower (a few minutes).

**Comment by rcousens** [comment]
Ok I've restored the cleanUpStaleJobs original query.

```
~/P/JMSJobQueueBundle ❯❯❯ phpunit
PHPUnit 4.7.6 by Sebastian Bergmann and contributors.

..........................................S.....

Time: 2.28 minutes, Memory: 48.00Mb

OK, but incomplete, skipped, or risky tests!
Tests: 48, Assertions: 141, Skipped: 1.
```

Works fine locally (I don't have PCNTL installed hence the skipped test), but the high concurrency test still fails on Travis.

**Pull request merged**